src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -496,6 +499,19 @@ private ImmutableArray<TExtension> GetAnalyzersForTypeNames(Assembly analyzerAss
 
                     Debug.Assert(type != null);
 
+                    // check if this references net framework, and issue a diagnostic if we don't allow that
+                    if (!_allowNetFramework)
+                    {
+                        var targetFrameworkAttribute = analyzerAssembly.GetCustomAttribute<TargetFrameworkAttribute>();
```

Comment:
`GetCustomAttribute<T>` will throw when dealing with invalid metadata (basically multiple copies of `TargetFarmeworkAttribute` even though it has `AllowMultiple = false`). Looked around a bit and seems like we are okay with this type of invalid IL breaking the compiler though. So think we can leave as is.

Reply:
I did look at the source code to see how it handled those cases and it seems to just take the first one, rather than throwing?

Reply:
The comment I saw said it threw. Either way though, seems like there is lots of precedenc for this. 

Mostyl I commented because it took me ten minutes to figure out where this call was coming from. I wanted to write something down to make me feel better about all the time I spent looking into it.